src/proto/Avalanche.proto

```diff
@@ -0,0 +1,198 @@
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+syntax = "proto3";
+
+package TW.Avalanche.Proto;
+option java_package = "wallet.core.jni.proto";
+
+message SECP256K1TransferOutput {
+    uint32 type_id = 1;            // 04 bytes
+    uint64 amount = 2;            // 08 bytes
+    uint64 locktime = 3;          // 08 bytes
+    uint32 threshold = 4;         // 04 bytes
+    repeated bytes addresses = 5; // 04 bytes + 20 bytes * len(addresses)
+}
+
+message SECP256K1MintOutput {
+    uint32 type_id = 1;            // 04 bytes
+    uint64 locktime = 2;          // 08 bytes
+    uint32 threshold = 3;         // 04 bytes
+    repeated bytes addresses = 4; // 04 bytes + 20 bytes * len(addresses)
+}
+
+message NFTTransferOutput {
+    uint32 type_id = 1;            // 04 bytes
+    uint32 group_id = 2;          // 04 bytes
+    bytes payload = 3;            // 04 bytes + len(payload)
+    uint64 locktime = 4;           // 08 bytes
+    uint32 threshold = 5;         // 04 bytes
+    repeated bytes addresses = 6; // 04 bytes + 20 bytes * len(addresses)
+}
+
+message NFTMintOutput {
+    uint32 typeID = 1;            // 04 bytes
+    uint32 group_id = 2;          // 04 bytes
+    uint64 locktime = 3;          // 08 bytes
+    uint32 threshold = 4;         // 04 bytes
+    repeated bytes addresses = 5; // 04 bytes + 20 bytes * len(addresses)
+}
+
+message TransactionOutput {
+    oneof output {
+        SECP256K1TransferOutput secp_transfer_output = 1;
+        SECP256K1MintOutput secp_mint_output = 2;
+        NFTTransferOutput nft_transfer_output = 3;
+        NFTMintOutput nft_mint_output = 4;
+    }
+}
+
+message SECP256K1TransferInput {
+    uint32 typeID = 1;                   // 04 bytes
+    uint64 amount = 2;                   // 08 bytes
+    repeated uint32 address_indices = 3; // 04 bytes + 04 bytes * len(address_indices)
+}
+
+message TransactionInput {
+    oneof input {
+        SECP256K1TransferInput secp_transfer_input = 1;
+    }
+}
+
+message SECP256K1MintOperation {
+    uint32 typeID = 1;                   // 4 bytes
+    repeated uint32 address_indices = 2; // 04 bytes + 04 bytes * len(address_indices)
+    SECP256K1MintOutput mint_output = 3;          // size(mint_output)
+    SECP256K1TransferOutput transfer_output = 4;  // size(transfer_output)
+}
+
+message NftMintOpOutput {
+    uint64 locktime = 1;
+    uint32 threshold = 2;
+    repeated bytes addresses = 3;
+}
+
+message NFTMintOp {
+    uint32 typeID = 1;                   // 04 bytes
+    repeated uint32 address_indices = 2; // 04 bytes + 04 bytes * len(address_indices)
+    uint32 group_id = 3;                 // 04 bytes
+    bytes payload = 4;                   // 04 bytes + len(payload)
+    repeated NftMintOpOutput outputs = 5;          // 04 bytes + size(outputs)
+}
+
+message NFTTransferOp {
```

Comment:
do we really need NFT transfers support here?

Reply:
I think so, collectible might be existed on both chains